### PR TITLE
[DeviceMesh] Removed unneeded `.to(cpu)`

### PR DESCRIPTION
**Pull request by awgu**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #124787
* #124780
* __->__ #124768
* #124767
* #124741
* #124651

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @penguinwu @fegin @XilunWu @wanchaol @fduwjj @wz337 @tianyu-l @wconstab @yf225 @chauhang @d4l3k

### CODE REVIEW

File: torch/distributed/device_mesh.py [hunk 1]
```diff
@@ -212,7 +212,7 @@ def __init__(
             if isinstance(mesh, torch.Tensor) and mesh.device.type != "cpu":
```
Comment (awgu) [inline torch/distributed/device_mesh.py:212]: We already check that the `mesh` tensor is on CPU. There is no need to include the `device="cpu"` in the `.to()` call.

Comment (wz337) [reply torch/distributed/device_mesh.py:212]: LGTM

**Review by wz337 — APPROVED** [review]

**Pull request closed without merging**